### ISSUE TITLE
Request Body Validation

### ISSUE THREAD

**Issue by GalaxyGorilla** [stoplightio/prism#167]
Hey, I just tested around with the mocking mode of prism and it turned out that JSON request bodies are not validated, e.g. I can just send whatever I want and get 200 with a response body. Do I overlook something obvious here to let the validation happen or is this just not implemented?

I used the petstore examples from swagger.

**Comment by tbarn** [comment]
Hi @GalaxyGorilla! Could you share more details of the requests you are making? There's a couple where it doesn't validate, but it should be validating for most.

**Comment by GalaxyGorilla** [comment]
I just used the [petstore minimal example](https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v2.0/yaml/petstore-simple.yaml) and then tried a POST:

```
 % curl -vvv localhost:4010/pets -X POST -d "some bullshit"
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 4010 (#0)
> POST /pets HTTP/1.1
> Host: localhost:4010
> User-Agent: curl/7.54.0
> Accept: */*
> Content-Length: 13
> Content-Type: application/x-www-form-urlencoded
>
* upload completely sent off: 13 out of 13 bytes
< HTTP/1.1 200 OK
< Content-Length: 69
< Content-Type: application/json
< Sl-Response-Valid: true
<
* Connection #0 to host localhost left intact
{"id":30459975,"name":"pariatur amet aliqua deserunt","tag":"mollit"}
```

My expectation was that the request body is validated against the parameter description in the spec.

**Comment by vidarw** [comment]
We would also like to see this feature implemented.
When a parameter required by the contract is missing, the expected behavior would be `400 Bad Request` preferably with an explanation of the missing parameters.

**Comment by tbarn** [comment]
One thing to watch out for is what is `additionalProperties` set to and what is the combination of `required` properties. I expanded on it here some: https://community.stoplight.io/t/contract-testing-not-working/295/7

**Comment by micahknox** [comment]
Same problem here. Here's my simple spec:
```
swagger: "2.0"
info:
  title: Sample API
  description: API description in Markdown.
  version: 1.0.0
host: webhook.site
basePath: /v1
schemes:
  - https
paths:
  /users:
    post:
      summary: Creates a new user.
      consumes:
        - application/json
      parameters:
        - in: body
          name: user
          description: The user to create.
          schema:
            type: object
            required:
              - userName
            properties:
              userName:
                type: string
              firstName:
                type: string
              lastName:
                type: string
      responses:
        201:
          description: Created
```
And my test request to a local prism instance running the validate command:
```
-> prism validate -d -s ./simple.oas2.yml
-> curl -vX POST http://localhost:4010/users
```
And here's the response:
```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 4010 (#0)
> POST /users HTTP/1.1
> Host: localhost:4010
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Transfer-Encoding: chunked
< Cache-Control: no-cache, private
< Content-Type: text/html; charset=UTF-8
< Date: Mon, 18 Feb 2019 22:40:36 GMT
< Server: nginx/1.10.3
< Sl-Request-Valid: true
< Sl-Response-Valid: true
< Vary: Accept-Encoding
```
I am expecting the 404 not found because of the host I am using but the `Sl-Request-Valid` header says my request is valid when I obviously did not post in a JSON object in the body. What am I missing?

**Comment by GabbyChang** [comment]
Same here. Seems not resolved for this issue. Any idea for this issue.

**Comment by tbarn** [comment]
Hey everyone! We just released a new patch version, 2.0.18. I'd recommend trying it with that. 

And for @micahknox can you try updating but add content-type header with `application/json` to see if that gets the contract testing working.
